test/grpc/integration/interceptor_headers.exs

```diff
@@ -0,0 +1,60 @@
+defmodule GRPC.Integration.EndpointTest do
+  use GRPC.Integration.TestCase
+
+  defmodule HelloServer do
+    use GRPC.Server, service: Helloworld.Greeter.Service
+
+    def say_hello(req, _stream) do
+      Helloworld.HelloReply.new(message: "Hello, #{req.name}")
+    end
+  end
+
+  defmodule AddHeadersClientInterceptor do
+    def init(label), do: label
+
+    def call(%{headers: headers} = stream, req, next, label) do
+      new_headers =
+        case Map.get(headers, "x-test-label") do
+          nil -> %{"x-test-label" => label}
+          original -> %{headers | "x-test-label" => "#{original} #{label}"}
+        end
+
+      new_stream = GRPC.Client.Stream.put_headers(stream, new_headers)
+      next.(new_stream, req)
+    end
+  end
+
+  defmodule HeadersServer do
+    def init(_), do: []
+
+    def call(req, stream, next, _) do
+      headers = GRPC.Stream.get_headers(stream)
+      %{name: original_name} = req
+      label = headers["x-test-label"]
+      updated = Helloworld.HelloRequest.new(name: "#{original_name} #{label}")
+      next.(updated, stream)
+    end
+  end
```

Comment:
Why is this interceptor is needed? I guess we only need verify headers in HelloServer?